src/EventStore.Core/Services/ElectionsService.cs

```diff
@@ -661,6 +722,7 @@ public class LeaderCandidate {
 			public readonly Guid InstanceId;
 			public readonly EndPoint HttpEndPoint;
 
+			public int ProposalNumber;
```

Comment:
@condron During testing, there are cases where ProposalNumber is having a value of 0 after elections are done. It is probably not being set when `LeaderCandidate` is created (since it's not part of the constructor). for safety we should pass `proposalNumber` in the constructor and make it immutable.

Reply:
Fixed